query_panel/main.js

```diff
@@ -31,9 +31,80 @@ const app = createApp({
       resizeTimer: undefined,
       windowHeight: DEFAULT_HEIGHT,
       scale: 1,
+      clipboardText: "",
     };
   },
   methods: {
+    // Converts the provided data to CSV format.
+    dataToCsv(data) {
+      try {
+        if (!data || data.length === 0) {
+          console.error("No data available to convert to CSV");
+          return "";
+        }
+        const replacer = (key, value) => (value === null ? "" : value);
+        const header = Object.keys(data[0]);
+        const csv = [
+          header.join(","),
+          ...data.map((row) =>
+            header
+              .map((fieldName) => {
+                let fieldData = row[fieldName];
+                if (fieldData && typeof fieldData === "string") {
+                  fieldData = fieldData.replace(/"/g, '""'); // Escape double quotes
+                  return `"${fieldData}"`; // Wrap in double quotes
+                }
+                return JSON.stringify(fieldData, replacer);
+              })
+              .join(","),
+          ),
+        ].join("\r\n");
+        return csv;
+      } catch (error) {
+        console.error("Error converting data to CSV:", error);
```

Comment:
Can we display an error to the user here. Otherwise the user has to switch to developer tools console to see why the button doesn't work.

II would also get rid of the additional throw.